utils.go

```diff
@@ -111,20 +109,25 @@ func newServicer(pairs ...typ.Pair) (srv *Service, err error) {
 		if err != nil {
 			return nil, err
 		}
+	case credential.ProtocolEnv:
```

Comment:
We need to support both env and other ways.

How about using `google.FindDefaultCredentials` instead?


ref: https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentials

Reply:
`FindDefaultCredentials` gets called automatically by the GCS client if no credentials are provided (e.g. if you don't use the `WithCredentialsJSON`).

So, if you pass `env` as credential type, the client will just fetch them all at runtime. 

This happens here: https://github.com/googleapis/google-api-go-client/blob/master/internal/creds.go#L50

Reply:
> gets called automatically by the GCS client if no credentials are provided

Yep, I know about this logic here.

But I don't want to depend on this behavior. I prefer to control them by ourselves.

- If the user uses `base64` or `file`, it's OK, read them and use `WithCredentialsJSON` to construct the client.
- If the user uses `env`, we use `FindDefaultCredentials` to read default credentials from the env.

Everything is clear, no need to read the code provided by deps.

What do you think?

Reply:
Absolutely got your point :) I cleaned up the code